homeassistant/components/input_select.py

```diff
@@ -44,44 +49,25 @@ def select_option(hass, entity_id, option):
 
 def setup(hass, config):
     """Setup input select."""
-    if not isinstance(config.get(DOMAIN), dict):
-        _LOGGER.error('Expected %s config to be a dictionary', DOMAIN)
-        return False
-
     component = EntityComponent(_LOGGER, DOMAIN, hass)
 
     entities = []
 
     for object_id, cfg in config[DOMAIN].items():
-        if object_id != slugify(object_id):
-            _LOGGER.warning("Found invalid key for boolean input: %s. "
-                            "Use %s instead", object_id, slugify(object_id))
-            continue
-        if not cfg:
-            _LOGGER.warning("No configuration specified for %s", object_id)
-            continue
-
         name = cfg.get(CONF_NAME)
         options = cfg.get(CONF_OPTIONS)
-
-        if not isinstance(options, list) or len(options) == 0:
-            _LOGGER.warning('Key %s should be a list of options', CONF_OPTIONS)
-            continue
-
-        options = [str(val) for val in options]
-
         state = cfg.get(CONF_INITIAL)
 
         if state not in options:
             state = options[0]
+            if state is not None:
+                _LOGGER.warning('Initial state %s is not in the options %s',
+                                state, ','.join(options))
 
         icon = cfg.get(CONF_ICON)
 
         entities.append(InputSelect(object_id, name, state, options, icon))
 
-    if not entities:
```

Comment:
Let's not remove this. An empty config should still be considered an error and we don't have to register services either ?